### ISSUE TITLE
Add Oklab colormode

### ISSUE THREAD

**Issue by Artoria2e5** [processing/p5.js#5164]
### How would this new feature help [increase access](https://github.com/processing/p5.js/blob/main/contributor_docs/access.md) to p5.js?
Having a more perceptual color system will help artists and anyone unfamiliar with the ideosyncracies of computer color space pick colors better.

### Most appropriate sub-area of p5.js?
- [x] Color

### New feature details:
[oklab](https://raphlinus.github.io/color/2021/01/18/oklab-critique.html) is a perceptual color space with a relatively simple transformation from linearized sRGB. Its cylindrical transformation, oklch (lightness-colorfulness-hue) would integrate well with the p5 color interface. Being perceptual, keeping the lightness fixed while changing the other two values really *does* keep the colors appear to be as bright.

### Notes for implementer
On the way to implementing oklab, p5 will have to implement linearized sRGB. This mode should also be exposed as a colormode for the sake of possible use in rendering.

Conversion from oklab to srgb may end up with colors that are out-of-gamut for sRGB, and it would be p5's choice to decide whether to clip the values at all.
* No-clip: A browser supporting a larger color-space may be able to process colors that are negative or higher than 255, but it one that only works in sRGB would simply clip the numbers.
* Clip: If p5 clips the number itself, it would have more control over [how to do it](https://bottosson.github.io/posts/gamutclipping/) and potentially give a better result, but at the same time degrading experience for wide-gamut systems.

It's also worth asking whether the other color models can handle out-of-range RGB values at all before deciding to no-clip.

**Comment by limzykenneth** [comment]
I think Oklab is too complicated and not mature enough to be included in p5.js at this point. It can be implemented as an addon library if desired though.

**Comment by Artoria2e5** [comment]
It's honestly the simpliest of "good" perceptual colorspaces. The more "mature" CIE L*a*b* has a similar (actually slightly more complex) construction, but its weakness in blue hues is more than obvious. The earlier attempts to "fix" CIE L*a*b* generally resulted in some even more complex things that I will not even dare to mention here. (IPT was a good model wrt hue, was of the same complexity, but Oklab does better with colorfulness.)

I might even claim that Oklab is simpler than HSL and HSB, as no "else if"s are needed in the calculation (you have inspired me to do a PR just to show how short the thing is). Maturity arguably doesn't matter either, because human eyes don't randomly change and give you new bugs or whatever.

**Comment by limzykenneth** [comment]
Considering p5.js itself doesn't have LAB color space in any of its form, the concern here is not about the quality of the color space or the simplicity of the implementation. What I mean by maturity has more to do with available resources for beginners to learn about any specific feature in p5.js (or anywhere else). From what you have linked I can only see very technical writings about the Oklab color space when many of p5.js' users may not even know what LAB color space are let alone understand why one is better than another.

Basically my argument here is not specifically about Oklab itself but rather the inclusion of LAB color space in general.

**Comment by Artoria2e5** [comment]
A picture of what the "rainbow" looks like is enough answer for "why oklch" for any beginner. The other concepts have very familiar analogs in HSB and HSL.

**Comment by Artoria2e5** [comment]
Ah you know what, chroma is an external library I can go for instead.
